packages/core/src/odata-common/selectable/field-builder.ts

```diff
@@ -0,0 +1,139 @@
+/* eslint-disable max-classes-per-file */
+
+import {
+  EdmTypeShared,
+  isOrderableEdmType,
+  OrderableEdmType
+} from '../edm-types';
+import { Constructable, Entity } from '../entity';
+import { ComplexTypeField } from './complex-type-field';
+import { EdmTypeField, EdmTypeForEdmOrFieldType } from './edm-type-field';
+import { OrderableEdmTypeField } from './orderable-edm-type-field';
+import { CollectionField, CollectionFieldType } from './collection-field';
+import { ConstructorOrField } from './constructor-or-field';
+import { FieldOptions } from './field';
+
+type ComplexTypeFieldConstructor<
+  ComplexTypeFieldT extends ComplexTypeField<
+    EntityT,
+    any,
+    NullableT,
+    SelectableT
+  >,
+  EntityT extends Entity,
+  NullableT extends boolean,
+  SelectableT extends boolean
+> = new (
+  fieldName: string,
+  fieldOf: ConstructorOrField<EntityT>,
+  fieldOptions?: FieldOptions<NullableT, SelectableT>
+) => ComplexTypeFieldT;
+
+export type IsSelectableField<T> = T extends Constructable<any> ? true : false;
+export type IsOrderableField<T> = T extends OrderableEdmType ? true : false;
+
+export class FieldBuilder<
+  EntityT extends Entity,
+  FieldOfT extends ConstructorOrField<EntityT>
+> {
+  constructor(public fieldOf: FieldOfT) {}
+
+  buildEdmTypeField<EdmT extends OrderableEdmType, NullableT extends boolean>(
```

Comment:
I think this is a matter of taste. You could have three different builders with a defining name but than you need in the generator invoke the fitting one. So I am fine with the overload. But could you add some small docs like: creates wither Edm or OrderableEdm depending on inptu.....

Reply:
But that would also mean, that you have to know which function to invoke when you use `Edm.String` vs. `Edm.Int16`.
I forgot the docs before and added them now.